In [3]:
import numpy as np
import pandas as pd
import  matplotlib.pyplot as plt
%matplotlib inline

# set up the data
data = pd.DataFrame()
data['kid'] = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby', 'Cindy', 'Oliver'] 
data['age'] = [14, 12, 11, 10, 8, 6, 8]


data.head(7)

kid  age
0    Greg   14
1  Marcia   12
2   Peter   11
3     Jan   10
4   Bobby    8
5   Cindy    6
6  Oliver    8

##### Question 1

###### MEAN

In [11]:
mean_age = sum(data['age']) / len(data['age'])
mean_age

9.857142857142858

In [12]:
np.mean(data['age'])

9.857142857142858

###### MEDIAN

visual inspection of data: median = 10

In [14]:
import statistics
statistics.median(data['age'])

10

In [15]:
np.median(data['age'])

10.0

###### MODE

visual inspection of data: mode = 8

In [16]:
statistics.mode(data['age'])

8

###### VARIANCE

In [19]:
data['age'].var()

7.476190476190475

In [20]:
np.var(data.age)

6.408163265306122

why is there a difference in calculated variance between pandas and numpy?

###### STANDARD DEVIATION

In [21]:
np.std(data['age'], ddof = 1)

2.734262327610589

###### STANDARD ERROR

In [25]:
np.std(data['age'], ddof = 1)/ np.sqrt(len(data['age']))

1.0334540197243192

##### Question 2

The estimate of central tendency I would choose would be the median (age = 10) as it best represents the range of the character's ages. 
The estimate of variance that is most descriptive of the ages respresnted in the variance (6.4 years). 

##### Question 3

In [26]:
data['age'] = [14, 12, 11, 10, 8, 7, 8]
data

kid  age
0    Greg   14
1  Marcia   12
2   Peter   11
3     Jan   10
4   Bobby    8
5   Cindy    7
6  Oliver    8

In [27]:
np.mean(data['age'])

10.0

In [28]:
np.median(data['age'])

10.0

In [29]:
statistics.mode(data['age'])

8

In [30]:
np.var(data.age)

5.428571428571429

In [31]:
np.std(data['age'], ddof = 1)

2.516611478423583

In [32]:
np.std(data['age'], ddof = 1)/ np.sqrt(len(data['age']))

0.9511897312113418

All estimates changed except for the median and the mode. Cindy is the youngest at age 6. When her age increased from 6 to 7, in reduced the range of ages, thus reducing the standard deviation, standard error, and the variance. The mean increased very slightly. Because her age was and remains the lowest, it did not affect the median or mode.

##### Question 4

In [37]:
data['kid'] = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby', 'Cindy', 'Jessica'] 
data['age'] = [14, 12, 11, 10, 8, 7, 1]
data

kid  age
0     Greg   14
1   Marcia   12
2    Peter   11
3      Jan   10
4    Bobby    8
5    Cindy    7
6  Jessica    1

In [34]:
np.mean(data['age'])

9.0

In [35]:
np.median(data['age'])

10.0

In [38]:
statistics.mode(data['age'])

StatisticsError: no unique mode; found 7 equally common values

There is no mode upon visual inspection of data. 

In [39]:
np.var(data.age)

15.428571428571429

In [40]:
np.std(data['age'], ddof = 1)

4.242640687119285

In [41]:
np.std(data['age'], ddof = 1)/ np.sqrt(len(data['age']))

1.6035674514745462

With the addition of 1 year old Jessica, the range of ages is much wider than previous. The median remains the same and does not reflect the larger age difference between the characters. A more descriptive estimate of cetral tendency would be the mean, to reflect the lowering of the average age by almost 1 full year. 
The most descriptive estimate of variance remains the variance, now increased to 15.4 years. 

##### Question 5

In [45]:
bb_fans = pd.DataFrame()
bb_fans['magazine'] = ['TV Guide', 'Entertainment Weekly', 'Pop Culture Today', 'SciPhi Phanatic'] 
bb_fans['percent_fans'] = [20, 23, 17, 5]


bb_fans.head()

magazine  percent_fans
0              TV Guide            20
1  Entertainment Weekly            23
2     Pop Culture Today            17
3       SciPhi Phanatic             5

In [46]:
np.mean(bb_fans['percent_fans'])

16.25

In [47]:
np.std(data['age'], ddof = 1)

4.242640687119285

In [48]:
np.std(data['age'], ddof = 1)/ np.sqrt(len(data['age']))

1.6035674514745462

In [ ]:
16.25% +/- 1.6%